constraints/src/main/java/gov/nasa/jpl/aerie/constraints/TypescriptCodeGenerationService.java

```diff
@@ -30,12 +30,16 @@ public static String generateTypescriptTypes(
     }
     result.add("}");
 
-    result.add("export type ResourceName = " + String.join(" | ", resources.keySet().stream().map($ -> "\"" + $ + "\"").toList()) + ";");
-    result.add("export type DiscreteResourceSchema<R extends ResourceName> =");
+    result.add("export type Resource = {");
     for (var resource: resources.keySet()) {
-      result.add(indent("R extends \"" + resource + "\" ? " + valueSchemaToTypescript(resources.get(resource)) + " :"));
+      result.add(indent("\"" + resource + "\": " + valueSchemaToTypescript(resources.get(resource)) + ","));
     }
-    result.add(indent("never;"));
+    result.add("}");
```

Comment:
lets add a semicolon here - I know they aren't strictly necessary, but it's best practice because of some weird stuff that _can_ happen without them.

Reply:
done